recipe/session/sessmodels/models.go

```diff
@@ -62,6 +67,67 @@ func (h *HandshakeInfo) SetJwtSigningPublicKeyList(updatedList []KeyInfo) {
 	h.rawJwtSigningPublicKeyList = updatedList
 }
 
+type GetJWKSFunction = func() (*keyfunc.JWKS, error)
+
+func GetJWKS() []GetJWKSFunction {
+	result := []GetJWKSFunction{}
+	corePaths := supertokens.GetAllCoreUrlsForPath("/.well-known/jwks.json")
+
+	for _, path := range corePaths {
+		result = append(result, func() (*keyfunc.JWKS, error) {
+			// RefreshUnknownKID - Fetch JWKS again if the kid in the header of the JWT does not match any in cache
+			// RefreshRateLimit - Only allow one re-fetch every 500 milliseconds
+			// RefreshInterval - Refreshes should occur every 600 seconds
+			jwks, err := keyfunc.Get(path, keyfunc.Options{
+				RefreshUnknownKID: true,
+				RefreshRateLimit:  time.Millisecond * time.Duration(JWKRefreshRateLimit),
+				RefreshInterval:   time.Millisecond * time.Duration(JWKCacheMaxAgeInMs),
+			})
+
+			return jwks, err
+		})
+	}
+
+	return result
+}
+
+/**
+This function is meant to fetch all available JWKs from all core instances and combine them into a sincle array.
+In most cases all core instances will return the same repsonse for the JWKS endpoint because they are all connected
+to the same database.
+
+Since the case where the same backend is connected to separate core instances with different databases connected
+to the instances is considered a very edge case situation, this function does not consider that scenario. So even
+the function logic does not combine anything as such, the final result is as if it did.
```

Comment:
```suggestion
This function fetches all JWKs from the first available core instance. This combines the other JWKS functions to become error resistant.

Every core instance a backend is connected to is expected to connect to the same database and use the same key set for token verification. Otherwise, the result of session verification would depend on which core is currently available.
```